### PR TITLE
Added logic for storing ETH2 deposits in DB and only request on-chain deposits further in time.

### PR DESCRIPTION
**Pull request by vnavascues**
Done:
- Extended functionality for storing on-chain requested deposits and only request new ones.
- Refactored existing tests, e.g. adding real timestamps, using `freezegun` (pip requirement) for faking current time and bypassing `datetime.now()`, etc.

Relates to #1721

### CODE REVIEW

File: rotkehlchen/chain/ethereum/eth2.py [hunk 1]
```diff
@@ -88,17 +148,108 @@ def get_eth2_staked_amount(
...
+        for address in new_addresses:
+            entry_name = f'{ETH2_DEPOSITS_PREFIX}_{address}'
+            database.update_used_query_range(
+                name=entry_name,
+                start_ts=ETH2_DEPLOYED_TS,
+                end_ts=to_ts,
+            )
+
+    # Get new deposits for existing addresses
+    if existing_addresses and min_from_ts <= to_ts:
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/eth2.py:213]: I think I had mentioned this in the standup too, but we **don't** want to rerun these kind of queries again and again even for a difference of 1 second. 

Either use `CACHE_RESPONSE_FOR_SECS` (10 mins for now) or use a new constant for 1 hour. So if saved data is older than an hour ... only then query again.


Do you think that makes sense?

File: rotkehlchen/tests/unit/test_eth2.py [hunk 2]
```diff
@@ -1,191 +1,387 @@
...
+        ethereum_manager_connect_at_start,
+        inquirer,
+        price_historian,
+        freezer,
+):
+    """
+    Test new on-chain requests for existing addresses start from last used
+    query range `end_ts`, to "now".
+
+    For avoiding request since ETH2 deposit contract deployment timestamp
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/test_eth2.py:288]: I don't think what you write here makes sense.

Do you simply want to say that you want to freeze `now` to be `1604506685` ?

**Review by LefterisJP — APPROVED** [review]
LGTM -- will also test along with the rest soon

File: rotkehlchen/tests/api/test_balances.py [hunk 3]
```diff
@@ -307,7 +307,12 @@ def test_query_all_balances_ignore_cache(
         msg = 'call count should increase since cache should have been ignored'
         assert all(fn.call_count == 2 for fn in function_call_counters), msg
         msg = 'etherscan call count should have doubled after forced token detection'
-        expected_count = full_query_etherscan_count * 2 - len(ethereum_accounts)
+        # NB: Subtract `get_eth2_staked_amount()` calls not done due to
+        # REQUEST_DELTA_TS constraint for cache ignoring request case.
+        eth2_calls_count = 14
```
Comment (vnavascues) [inline rotkehlchen/tests/api/test_balances.py:312]: @LefterisJP Please don't think I'm happy about this hard coded number. I'm aware there are two addresses and I've followed the calls but I'm still unsure if there is a linear relation between number of addresses and calls (e.g. 1 address = 7 calls, 2 addresses = 14 calls, etc.).
I've tried changing the parametrized `number_of_eth_accounts` but the test fails for other reasons.
Feel free to change this statement for something more meaningful!

Comment (LefterisJP) [reply rotkehlchen/tests/api/test_balances.py:312]: no worries victor, we can figure it out later. Thanks a lot for looking at this. All is green :)

**Review by LefterisJP — APPROVED** [review]
LGTM

**Pull request merged**